### ISSUE TITLE
[Question]We have IsDefined already, why we need IsOptional?

### ISSUE THREAD

**Issue by Diluka** [pleerock/class-validator#120]
1. `@IsDefined()` checks the property must have a value rather than `null` or `undefined`
2. if put `@IsDefined()` and `@IsOptional()` together, what is the meaning?

seems `@IsOptional()` doesn't work properly

**Comment by 19majkel94** [comment]
> We have IsDefined already, why we need IsOptional?

When you have 5 must have properties and one is optional, it's better to mark the one with `@IsOptional` or set the 5 with `@IsDefined()` and set `skipMissingProperties`?

**Comment by Diluka** [comment]
```ts
class C {
    @IsOptional()
    @IsInt()
    age: number;
}

validate(new C()).then((errors) => {
    console.log(errors);
});
```
output:
```
[ ValidationError {
    target: C {},
    value: undefined,
    property: 'age',
    children: [],
    constraints: { isInt: 'age must be an integer number' } } ]
```

Is this right? So, I confuse about how `@IsOptional()` working.

**Comment by 19majkel94** [comment]
https://github.com/pleerock/class-validator/pull/110

**Comment by Diluka** [comment]
I checked this PR before. I think it already contains in the latest release.
You mean it has been fixed, but haven't been released yet?

**Comment by 19majkel94** [comment]
Yup!

**Comment by Diluka** [comment]
close, released #116
